packages/analytics/src/pingback.ts

```diff
@@ -1,61 +1,84 @@
+const getRandomId = () => {
+    // it exists in memory
+    if (!gl.giphyRandomId) {
+        try {
+            // it exists in storage
+            gl.giphyRandomId = localStorage.getItem('giphyRandomId')
+        } catch (_) {}
+        if (!gl.giphyRandomId) {
+            // we need to create it
+            gl.giphyRandomId = uuid()
+            try {
+                // save in storage
+                localStorage.setItem('giphyRandomId', gl.giphyRandomId)
+            } catch (_) {}
+        }
+    }
+    return gl.giphyRandomId
+}
 
 let loggedInUserId = ''
 
-function fetchPingbackRequest() {
-    forEach(queuedPingbacks, (actionMap: ActionMap, pingbackType: PingbackEventType) => {
-        if (actionMap) {
-            forEach(actionMap, (action: PingbackRequestAction[], responseId: string) => {
-                // if there are no actions lined up inside this pingbackType do nothing
-                if (action.length) {
-                    const session = createSession(pingbackType, action, responseId, loggedInUserId)
-                    sendPingback(session)
-                    // empty this specific batch
-                    actionMap[responseId] = []
-                }
-            })
-        }
-    })
+function sendPingbacks() {
+    const sendEvents = [...queuedPingbackEvents]
+    queuedPingbackEvents = []
+    sendPingback(sendEvents)
 }
 
-const debouncedPingbackEvent = debounce(1000, fetchPingbackRequest)
+const debouncedPingbackEvent = debounce(1000, sendPingbacks)
+
+const pingback = ({ gif, userId, eventType, actionType, attributes, queueEvents = true }: Pingback) => {
+    // save the user id for whenever create session is invoked
+    loggedInUserId = userId ? String(userId) : loggedInUserId
+
+    /* istanbul ignore next */
+    // get the giphy_pbid cookie
+    const user_id = cookie.parse(document ? document.cookie : ({} as any)).giphy_pbid
 
-const pingback = ({ gif, user, responseId, type: pingbackType, actionType, position, attributes }: Pingback) => {
-    // not all endpoints provide a response_id
-    if (!responseId) {
-        Logger.debug(`Pingback aborted for ${gif.id}, no responseId`)
-        return
+    const newEvent: PingbackEvent = {
+        ts: Date.now(),
+        attributes,
+        action_type: actionType,
     }
-    const { id, bottle_data = {} } = gif
-    const { tid } = bottle_data
 
-    // save the user id for whenever create session is invoked
-    loggedInUserId = user && user.id ? String(user.id) : loggedInUserId
+    if (loggedInUserId) {
+        newEvent.logged_in_user_id = loggedInUserId
+    }
 
-    // the queue doesn't exist for this pingbackType yet so create it
-    if (!queuedPingbacks[pingbackType]) queuedPingbacks[pingbackType] = {}
+    if (gif) {
+        if (!gif.analytics_response_payload) {
+            // abort pingback, analytics_response_payload is required for gif events
+            return
+        }
+        const gifEvent = newEvent as PingbackGifEvent
```

Comment:
pingbacks in the SDK don't require a gif object, this will let us leverage this code on O&O for some of the custom pingbacks